controllers/ltpa_keys_sharing.go

```diff
@@ -0,0 +1,315 @@
+				r.DeleteResource(ltpaKeysCreationScriptConfigMap)
+			}
+			if err != nil && kerrors.IsNotFound(err) {
+				ltpaKeysCreationScriptConfigMap.Data = make(map[string]string)
+				script, err := ioutil.ReadFile("utils/create_ltpa_keys.sh")
+				if err != nil {
+					return err, ""
+				}
+				ltpaKeysCreationScriptConfigMap.Data["create_ltpa_keys.sh"] = string(script)
+				r.CreateOrUpdate(ltpaKeysCreationScriptConfigMap, instance, func() error {
+					return nil
+				})
+			}
+
+			// Verify the utils/create_ltpa_keys.sh script has been loaded before starting the LTPA Job
+			err = r.GetClient().Get(context.TODO(), types.NamespacedName{Name: ltpaKeysCreationScriptConfigMap.Name, Namespace: ltpaKeysCreationScriptConfigMap.Namespace}, ltpaKeysCreationScriptConfigMap)
+			if err == nil {
+				// Run the Kubernetes Job to generate the shared ltpa.keys file and LTPA Secret
+				err = r.GetClient().Get(context.TODO(), types.NamespacedName{Name: generateLTPAKeysJob.Name, Namespace: generateLTPAKeysJob.Namespace}, generateLTPAKeysJob)
+				if err != nil && kerrors.IsNotFound(err) {
+					err = r.CreateOrUpdate(generateLTPAKeysJob, instance, func() error {
+						lutils.CustomizeLTPAJob(generateLTPAKeysJob, instance, ltpaSecret.Name, ltpaServiceAccountName, ltpaKeysCreationScriptConfigMap.Name)
+						return nil
+					})
+					if err != nil {
+						return fmt.Errorf("Failed to create Job " + generateLTPAKeysJob.Name), ""
+					}
+				} else if err != nil {
+					return fmt.Errorf("Failed to get Job " + generateLTPAKeysJob.Name), ""
+				}
+			}
+		}
+
+		// Reconcile the Job
+		err = r.GetClient().Get(context.TODO(), types.NamespacedName{Name: generateLTPAKeysJob.Name, Namespace: generateLTPAKeysJob.Namespace}, generateLTPAKeysJob)
+		if err != nil && kerrors.IsNotFound(err) {
+			return fmt.Errorf("Waiting for Job " + generateLTPAKeysJob.Name + " to be created."), ""
+		} else if err != nil {
+			return fmt.Errorf("Failed to get Job " + generateLTPAKeysJob.Name), ""
+		}
+		if len(generateLTPAKeysJob.Status.Conditions) > 0 && generateLTPAKeysJob.Status.Conditions[0].Type == v1.JobFailed {
+			return fmt.Errorf("Job " + generateLTPAKeysJob.Name + " has failed. Manually clean up hung resources by setting .spec.manageLTPA to false in the " + ltpaKeySharingLeaderName + " instance."), ""
+		}
+		return fmt.Errorf("Waiting for Job " + generateLTPAKeysJob.Name + " to be completed."), ""
+	} else if err != nil {
+		return err, ""
+	} else {
+		if !isLTPAKeySharingLeader {
+			return nil, ltpaSecret.Name
+		}
+	}
+
+	// The LTPA Secret is created (in other words, the LTPA Job has completed), so delete the Job request
+	err = r.DeleteResource(ltpaJobRequest)
+	if err != nil {
+		return err, ltpaSecret.Name
+	}
+
+	// Create the Liberty Server XML ConfigMap if it doesn't exist
+	configMapErr := r.GetClient().Get(context.TODO(), types.NamespacedName{Name: ltpaServerXMLConfigMap.Name, Namespace: ltpaServerXMLConfigMap.Namespace}, ltpaServerXMLConfigMap)
```

Comment:
Since the encrypted password is included in the config, perhaps it's better to use Secret for server config as well

Reply:
I changed this to a Secret now